dex-template/__DexName__-integration.test.ts.template

```diff
@@ -36,21 +37,89 @@ const TokenB = Tokens[network][TokenBSymbol];
 
 const amounts = [0n, BI_POWS[18], 2000000000000000000n];
 
+const dexHelper = new DummyDexHelper(network);
 const dexKey = '__DexName__';
 
+function getReaderCalldata(
+  exchangeAddress: string,
+  readerIface: Interface,
+  amounts: bigint[],
+  funcName: string,
+  // TODO: Put here additional arguments you need
+) {
+  return amounts.map(amount => ({
+    target: exchangeAddress,
+    callData: readerIface.encodeFunctionData(funcName, [
+      // TODO: Put here additional arguments to encode them
+      amount,
+    ]),
+  }));
+}
+
+function decodeReaderResult(
+  results: Result,
+  readerIface: Interface,
+  funcName: string,
+) {
+  // TODO: Adapt this function for your needs
+  return results.map(result => {
+    const parsed = readerIface.decodeFunctionResult(funcName, result);
+    return BigInt(parsed[0]._hex);
+  });
+}
+
+async function checkOnChainPricing(
+  __DexNameCamel__: __DexName__,
+  funcName: string,
```

Comment:
not sure why you would pass funcName

Reply:
Because we need different function names for different scenarios. It may be: `sellBase` or `sellQuote`